assignment/models/homework.py

```diff
@@ -143,3 +157,11 @@ def reassign_assignment(self):
         self.ensure_one()
         self.state = 'active'
         return True
+
+    @api.multi
+    def unlink(self):
+        for rec in self:
+            if rec.state != 'draft':
+                raise ValidationError(_('''You can delete record
+                                        in draft state only.'''))
+            return super(SchoolStudentAssignment, self).unlink()
```

Comment:
shift left. it should not be inside loop

Reply:
done